onchain/cairo/src/afk_id/nameservice.cairo

```diff
@@ -318,22 +355,152 @@ pub mod Nameservice {
+            assert(
+                existing_auction.minimal_price == 0, 
+                UserNameClaimErrors::AUCTION_DOESNT_EXIST
+            );
+
+            // Create a new auction
+            let new_auction = Auction {
+                owner: caller_address,
+                minimal_price: minimal_price,
+                is_accepted_price_reached: is_accepted_price_reached,
+                highest_bid: 0,
+                highest_bidder: contract_address_const::<0>(),
+            };
+
+            // Store the auction
+            self.auctions.write(username, new_auction);
+            self
+                .emit(
+                    AuctionCreated {
+                        owner: caller_address,
+                        username: username,
+                        minimal_price,
+                        is_accepted_price_reached
+                    }
+                );
+        }
 
         // TODO
-        fn place_order(ref self: ContractState, username: felt252, amount: u256) {}
+        fn place_order(ref self: ContractState, username: felt252, amount: u256) {
+            let auction = self.auctions.read(username);
+            assert(
+                auction.minimal_price > 0, 
+                UserNameClaimErrors::AUCTION_DOESNT_EXIST
+            );
+            assert(amount > auction.highest_bid, UserNameClaimErrors::BID_LOW);
+
+            // Create a new order
+            let bidder = get_caller_address();
+            let quote_token = self.token_quote.read();
+
+             // check if new bidder already has an outbidded amount for the username(still in the contract)
+            let bidder_amount = self.order_return.entry(bidder).entry(username).read();
+            self.order_return.entry(bidder).entry(username).write(0);
+
+            if amount > bidder_amount {
+                if self.is_payment_enabled.read() {
+                    let payment_token = IERC20Dispatcher { contract_address: quote_token };
+                    payment_token.transfer_from(bidder, get_contract_address(), (amount - bidder_amount));
+                }
+            }
+
+            let order_id = self.order_count.read(username) + 1;
+            let new_order = Order { bidder: bidder, amount: amount };
 
+            let old_order = self.orders.read(username);
+
+            // Store the order
+            self.orders.write(username, new_order);
+            self.order_count.write(username, order_id);
+
+            let order_return_amount = self.order_return.entry(old_order.bidder).entry(username).read();
+            self.order_return.entry(old_order.bidder).entry(username).write((order_return_amount + old_order.amount));
+
+            // Update auction if this is the highest bid
+            let mut updated_auction = auction;
+            updated_auction.highest_bid = amount;
+            updated_auction.highest_bidder = bidder;
+            updated_auction.is_accepted_price_reached = true;
```

Comment:
Not sure why we change the is_accepted_price_reached here, don't think we need and can cause issue